File: PriorAuthApi/Entities/AuthRule.cs
namespace PriorAuthApi.Entities
{
    public class AuthRule
    {
        public int Id { get; set; }
        public RequestType RequestType { get; set; }
        public required string CodeSystem { get; set; }  // e.g., "CPT", "HCPCS"
        public required string Code { get; set; }
        public string? IndicationCode { get; set; }  // ICD-10 code for the indication
        public required string DisplayName { get; set; }
        public string? IndicationDisplayName { get; set; }
        public required string FormDefinition { get; set; }       // JSON array
        public required string RuleDefinition { get; set; }       // JSON array
        public bool IsActive { get; set; }
        public DateOnly EffectiveDate { get; set; }
        public DateOnly CreatedAt { get; set; } = DateOnly.FromDateTime(DateTime.Now);
    }
}

File: PriorAuthWeb/src/types/AuthRule.ts
export type FormFieldType = "boolean" | "select" | "number" | "text";

export interface FormField {
    name: string;
    label: string;
    type: FormFieldType;
    options?: string[];
    validation: ValidationRules;
    defaultValue?: string | number;
    editable?: boolean;
}

export interface ValidationRules {
    required: boolean;
    maxLength?: number;
    min?: number;
    max?: number;
    integer?: boolean;
}

export interface FormDefinition {
    fields: FormField[];
    medicationFields?: FormField[]; // Optional fields specific to medication requests
}

export interface AuthRule {
    id: number;
    requestType: "Service" | "Medication";
    codeSystem: string; // e.g. "HCPCS"
    code: string; // e.g. "J0135"
    indicationCode: string; // e.g. "M06.9"
    displayName: string; // e.g. "Humira (adalimumab)"
    formDefinition: FormDefinition;
    ruleDefinition: Record<string, string | number | boolean>;
    isActive: boolean;
    effectiveDate: string; // ISO date string
    createdAt: string; // ISO date string
}